### ISSUE TITLE
Correct format of configuration file

### ISSUE THREAD

**Issue by D19cannon** [lingui/js-lingui#557]
Hi guys,

I have been trying to set up Lingui for two days now. While following the documentation and a few tutorials I am not able to solve a few errors that are occurring.

I tried to set it up locally and globally many times. The Lingui command is running and working, however, the configurations are always a mess. Whenever I am able to solve one another simply just shows up.

For example: 
I had that lingui is not a valid option or something when I am extracting.
add-locale is not a function.
babel does not recognise this function etc.
Sometimes it extracts and I get the JSON messages, but it won't update the <Trans> target with the correct language or just displays the id.

The following error for example occurs:
![Screenshot 2019-08-28 at 10 49 38](https://user-images.githubusercontent.com/14916043/63836471-4a3db400-c982-11e9-9c9f-75a78d458baf.png)

With the following standard configuration:
`{
   "lingui": {
    "compileNamespace": "cjs",
    "extractBabelOptions": {},
    "fallbackLocale": "fi",
    "sourceLocale": "fi",
    "localeDir": "<rootDir>/locale",
    "srcPathDirs": [
        "<rootDirsrc"
    ],
    "srcPathIgnorePatterns": [
        "/node_modules/"
    ],
    "format": "lingui",
  }
}`

Long story short: **is the documentation properly up to date? and also would it be possible to add the correct version that are required to install everything in the documentation.**

**Comment by tricoder42** [comment]
It looks like you're using configuration file (e.g. `lingui.config.js` or `.linguirc`). In such case, remove the `lingui` key and use just the configuration options:

```
// .linguirc
{ 
  "compileNamespace": "cjs", 
  "extractBabelOptions": {}, 
  "fallbackLocale": "fi", 
  "sourceLocale": "fi", 
  "localeDir": "<rootDir>/locale", 
  "srcPathDirs": [ "<rootDirsrc" ], 
  "srcPathIgnorePatterns": [ "/node_modules/" ], 
  "format": "lingui"
}
```

If you had configuration in `package.json`, you would have to put it under `lingui` key. Does it make sense?

**Comment by D19cannon** [comment]
Hey @tricoder42 thanks for your response!
In tried it earlier by removing that, but then nothing will be build in my message.json

I just removed it again to get the error and the following shows up:

`─daniel@DonBranco /Users/daniel/sites/konkka_v2  ‹system›  <master*>
╰─$ npm run extract

> konkka@0.1.0 extract /Users/daniel/sites/konkka_v2
> lingui extract

● Deprecation Warning:

 Option localeDir is deprecated. Configure source paths using catalogs instead.

    @lingui/cli now treats your current configuration as:

    {
      "catalogs": [
  {
    "path": "<rootDir>/locale/{locale}/messages",
    "include": [
      "<rootDir>"
    ],
    "exclude": [
      "*/node_modules/*"
    ]
  },
  {
    "path": "<rootDir>/src/locale/{locale}/messages",
    "include": [
      "<rootDir>/src"
    ],
    "exclude": [
      "/node_modules/"
    ]
  }
]
    }

    Please update your configuration.


Documentation: https://lingui.js.org/ref/conf.html
● Deprecation Warning:

 Option srcPathDirs is deprecated. Configure source paths using catalogs instead.

    @lingui/cli now treats your current configuration as:

    {
      "catalogs": [
  {
    "path": "<rootDir>/locale/{locale}/messages",
    "include": [
      "<rootDir>"
    ],
    "exclude": [
      "*/node_modules/*"
    ]
  },
  {
    "path": "<rootDir>/src/locale/{locale}/messages",
    "include": [
      "<rootDir>/src"
    ],
    "exclude": [
      "/node_modules/"
    ]
  },
  {
    "path": "<rootDir>/src/locale/{locale}/messages",
    "include": [
      "<rootDir>/src"
    ],
    "exclude": [
      "/node_modules/"
    ]
  }
]
    }

    Please update your configuration.


Documentation: https://lingui.js.org/ref/conf.html
● Deprecation Warning:

 Option srcPathIgnorePatterns is deprecated. Configure excluded source paths using catalogs instead.

    @lingui/cli now treats your current configuration as:

    {
      "catalogs": [
  {
    "path": "<rootDir>/locale/{locale}/messages",
    "include": [
      "<rootDir>"
    ],
    "exclude": [
      "*/node_modules/*"
    ]
  },
  {
    "path": "<rootDir>/src/locale/{locale}/messages",
    "include": [
      "<rootDir>/src"
    ],
    "exclude": [
      "/node_modules/"
    ]
  },
  {
    "path": "<rootDir>/src/locale/{locale}/messages",
    "include": [
      "<rootDir>/src"
    ],
    "exclude": [
      "/node_modules/"
    ]
  },
  {
    "path": "<rootDir>/src/locale/{locale}/messages",
    "include": [
      "<rootDir>/src"
    ],
    "exclude": [
      "/node_modules/"
    ]
  }
]
    }

    Please update your configuration.


Documentation: https://lingui.js.org/ref/conf.html
No locales defined!

Add 'locales' to your configuration. See https://lingui.js.org/ref/conf.html#locales
Cannot process file Expected atom at position 2
    \p{Script_Ex
      ^
Cannot process file Expected atom at position 2
    \p{ASCII_Hex
      ^
Cannot process file Expected atom at position 2
    \p{Regional_
      ^
Cannot process file Expected atom at position 2
    \p{Script_Ex
      ^
Cannot process file Expected atom at position 2
    \p{Script_Ex
      ^
Cannot process file Expected atom at position 2
    \p{ASCII_Hex
      ^
Cannot process file Expected atom at position 2
    \p{Regional_
      ^
Cannot process file Expected atom at position 2
    \p{Script_Ex
      ^
Cannot process file Expected atom at position 2
    \p{Script_Ex
      ^
Cannot process file /Users/daniel/sites/konkka_v2/node_modules/@babel/preset-typescript/test/fixtures/flow-compat/js-invalid/input.js: Unexpected reserved word 'enum' (1:0)

> 1 | enum Example {
    | ^
  2 |   Value
  3 | }
  4 |
Cannot process file /Users/daniel/sites/konkka_v2/node_modules/@babel/preset-typescript/test/fixtures/flow-compat/ts-invalid/input.ts: Unexpected token (1:1)

> 1 |  || ;
    |  ^
  2 | ;
  3 | foo;
  4 |
Cannot process file Cannot find module 'babel-plugin-syntax-jsx'
Cannot process file Cannot find module 'babel-plugin-syntax-jsx'
Cannot process file /Users/daniel/sites/konkka_v2/node_modules/@babel/preset-typescript/test/fixtures/jsx-compat/ts-invalid/input.ts: Unterminated regular expression (1:1)

> 1 | />);;
    |  ^
  2 |
Cannot process file Cannot find module 'babel-plugin-syntax-jsx'
[BABEL] Note: The code generator has deoptimised the styling of /Users/daniel/sites/konkka_v2/node_modules/@material-ui/core/umd/material-ui.development.js as it exceeds the max of 500KB.
Cannot process file /Users/daniel/sites/konkka_v2/node_modules/@storybook/addon-actions/src/lib/errors/index.js: Support for the experimental syntax 'exportDefaultFrom' isn't currently enabled (1:8):

> 1 | export DecycleError from './DecycleError';
    |        ^
  2 |

Add @babel/plugin-proposal-export-default-from (https://git.io/vb4yH) to the 'plugins' section of your Babel config to enable transformation.
Cannot process file /Users/daniel/sites/konkka_v2/node_modules/@storybook/addon-actions/src/lib/index.js: Support for the experimental syntax 'exportDefaultFrom' isn't currently enabled (2:8):

  1 | export const CYCLIC_KEY = '$___storybook.isCyclic';
> 2 | export decycle from './decycle';
    |        ^
  3 | export retrocycle from './retrocycle';
  4 | export reviver from './reviver';
  5 |

Add @babel/plugin-proposal-export-default-from (https://git.io/vb4yH) to the 'plugins' section of your Babel config to enable transformation.
Cannot process file /Users/daniel/sites/konkka_v2/node_modules/@storybook/addon-actions/src/lib/util/index.js: Support for the experimental syntax 'exportDefaultFrom' isn't currently enabled (1:8):

> 1 | export canConfigureName from './canConfigureName';
    |        ^
  2 | export getPropertiesList from './getPropertiesList';
  3 | export isObject from './isObject';
  4 | export muteProperty from './muteProperty';

Add @babel/plugin-proposal-export-default-from (https://git.io/vb4yH) to the 'plugins' section of your Babel config to enable transformation.
Cannot process file /Users/daniel/sites/konkka_v2/node_modules/@storybook/addon-links/src/react/index.js: Support for the experimental syntax 'exportDefaultFrom' isn't currently enabled (1:8):

> 1 | export default from './components/link';
    |        ^
  2 |

Add @babel/plugin-proposal-export-default-from (https://git.io/vb4yH) to the 'plugins' section of your Babel config to enable transformation.
Cannot process file /Users/daniel/sites/konkka_v2/node_modules/@storybook/addons/src/index.js: Support for the experimental syntax 'exportDefaultFrom' isn't currently enabled (5:8):

  3 | // import { TabWrapper } from '@storybook/components';
  4 |
> 5 | export mockChannel from './storybook-channel-mock';
    |        ^
  6 | export { makeDecorator } from './make-decorator';
  7 |
  8 | export class AddonStore {

Add @babel/plugin-proposal-export-default-from (https://git.io/vb4yH) to the 'plugins' section of your Babel config to enable transformation.
Cannot process file EISDIR: illegal operation on a directory, read
`

**Comment by D19cannon** [comment]
After removing all the imports, components, conf etc.

The following message start to occur (to be clear I am not using lingui any where)

![Screenshot 2019-08-28 at 12 30 44](https://user-images.githubusercontent.com/14916043/63843927-e4582900-c98f-11e9-942e-3b3732886b1c.png)

**Comment by tricoder42** [comment]
Please create a repository with minimal example. It's hard to debug it without looking at the source files. Also, take a look at `examples` directly in this repository. It contains working examples with correct configuration.

**Comment by D19cannon** [comment]
No worries.
I deleted everything again and checked all the versions etc. 

Got it somehow up and working. There was one minor thing that I noticed with the whole setup.

When using add-locale it creates a "locales" folder and in the documentation you use: locale/
https://lingui.js.org/tutorials/react.html

I am not entirely sure how I solved it. But starting over from a clean point and redo everything worked for me.

**Comment by D19cannon** [comment]
It is easy to make mistakes when people simply copy and paste your code.

[https://lingui.js.org/tutorials/react.html](url)

Also, might be nice to add some version requirements.
Such as, react-scripts v^3.0.0

**Update these lines in your documentations!!** 
---------------------------------------------------
**Current:**
import catalogCs from './locale/cs/messages.js'

**Should be:**
import catalogCs from './locales/cs/messages.js'

---------------------------------------------------
**Current:**
Let’s take a look at file locale/cs/messages.json

**Should be:**
Let’s take a look at file locales/cs/messages.json

----------------------------------------------------
**Current:**
If you look inside locale directory

**should be:**
If you look inside locals directory

---------------------------------------------------

**Comment by tricoder42** [comment]
Thanks for suggestions 👍  Feel free to send a PR. You can edit the file directly on GitHub without cloning the repo: https://github.com/lingui/js-lingui/edit/master/docs/tutorials/react.rst
